src/utils/getActiveElement.ts

```diff
@@ -1,31 +1,19 @@
-const getNestedShadowActiveElement = (shadowRoot: ShadowRoot): HTMLElement | undefined =>
-  shadowRoot.activeElement
-    ? shadowRoot.activeElement.shadowRoot
-      ? getNestedShadowActiveElement(shadowRoot.activeElement.shadowRoot)
-      : (shadowRoot.activeElement as HTMLElement)
-    : undefined;
-
-const getNestedIframeActiveElement = (iframe: HTMLIFrameElement): HTMLElement | undefined => {
-  const activeElement = iframe?.contentWindow?.document?.activeElement;
-
-  return activeElement
-    ? activeElement instanceof HTMLIFrameElement
-      ? getNestedIframeActiveElement(activeElement)
-      : (activeElement as HTMLElement)
-    : undefined;
-};
-
 /**
  * returns active element from document or from nested shadowdoms
  */
-export const getActiveElement = (): HTMLElement | undefined => {
+
+export const getActiveElement = (inDocument: Document | ShadowRoot | undefined = document): HTMLElement | undefined => {
+  if (!inDocument || !inDocument.activeElement) {
+    return undefined;
+  }
+
+  const { activeElement } = inDocument;
+
   return (
-    document.activeElement
-      ? document.activeElement.shadowRoot
-        ? getNestedShadowActiveElement(document.activeElement.shadowRoot)
-        : document.activeElement instanceof HTMLIFrameElement
-        ? getNestedIframeActiveElement(document.activeElement)
-        : document.activeElement
-      : undefined
-  ) as any; // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    activeElement.shadowRoot
+      ? getActiveElement(activeElement.shadowRoot)
+      : activeElement instanceof HTMLIFrameElement && activeElement.contentWindow?.document
+      ? getActiveElement(activeElement.contentWindow.document)
+      : activeElement
+  ) as HTMLElement;
```

Comment:
now there is no need for separate functions to dive deeper - function just shifts from one document to another.